entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -557,10 +567,184 @@ class EntityService(
     }
 
     @Transactional
-    fun updateEntityAttribute(id: URI, attribute: String, payload: String, contexts: List<String>): Int {
+    fun partialUpdateEntityAttribute(
+        entityId: URI,
+        attribute: String,
+        payload: Map<String, List<Any>>,
+        contexts: List<String>
+    ): Int {
         val expandedAttributeName = expandJsonLdKey(attribute, contexts)!!
-        val attributeValue = parseJsonLdFragment(payload)["value"]!!
-        return neo4jRepository.updateEntityAttribute(id, expandedAttributeName, attributeValue)
+        val datasetId = payload.getDatasetId()
+        return when {
+            neo4jRepository.hasRelationshipOfType(EntitySubjectNode(entityId), attribute.toRelationshipTypeName()) ->
```

Comment:
if you're working with the compact form, you don't need to call `toRelationshipTypeName()` (but that may change if you expand the attribute name in the handler)